### PR TITLE
docs(flutter): Document `Sentry.runZonedGuarded`

### PR DESCRIPTION
**Pull request by denrase**
Document `Sentry.runZonedGuarded`

## IS YOUR CHANGE URGENT?  

Help us prioritize incoming PRs by letting us know when the change needs to go live.
- [ ] Urgent deadline (GA date, etc.): 
- [ ] Other deadline: 
- [x] None: Not urgent, can wait up to 1 week+

## Related

- [Support custom Sentry.runZoneGuarded zone creation](https://github.com/getsentry/sentry-dart/pull/2088)

### CODE REVIEW

File: platform-includes/capture-error/flutter.mdx
Comment (buenaflor) [inline platform-includes/capture-error/flutter.mdx:25]: I wonder if we should also include a tab in the code snippet like `FlutterWeb with WidgetsBindingIntegration` or something like that in the getting started page because I think it's rather common to want to initialize the WidgetsBinding before the Sentry init and even though we add the info here, I would imagine it's still hard to find, the user would have to scroll through all the pages

but not sure

**Review by coolguyzone — APPROVED** [review]
Thanks for adding! Looks good to me, we can merge this pending @buenaflor's approval. 🏄‍♂️

File: platform-includes/capture-error/flutter.mdx
Comment (denrase) [reply platform-includes/capture-error/flutter.mdx:25]: Added it as a tab, not sure if we should add additional reference in the text, as it is also describe here in detail.

File: docs/platforms/flutter/index.mdx
Comment (buenaflor) [inline docs/platforms/flutter/index.mdx:83]: maybe explain why to use this method e.g in Flutter Web if you want to use WidgetsBindingIntegration before SentryFlutter.init

File: docs/platforms/flutter/index.mdx
Comment (buenaflor) [inline docs/platforms/flutter/index.mdx:73]: is it possible to make the tab title shorter somehow hmm

File: platform-includes/capture-error/flutter.mdx
Comment (buenaflor) [inline platform-includes/capture-error/flutter.mdx:9]: doesn't the runZonedGuarded only happen on Web?

Comment (denrase) [reply platform-includes/capture-error/flutter.mdx:9]: Afaik there was no onError callback on version with Flutter < 3.3, so we'd also use it there.

File: docs/platforms/flutter/index.mdx
Comment (denrase) [reply docs/platforms/flutter/index.mdx]: Wrote a bit long winded explanation, maybe you can add your 2 cents to make it more concise.

**Comment by denrase** [comment]
@buenaflor Vercel run pls. :)

**Comment by buenaflor** [comment]
@kahest cc

**Review by buenaflor — COMMENTED** [review]
I'll have another look in the vercel preview and if it looks good I'll approve

**Pull request merged**
